sympy/core/mul.py

```diff
@@ -1264,26 +1266,38 @@ def _eval_is_odd(self):
         is_integer = self.is_integer
 
         if is_integer:
-            r, acc = True, 1
+            #The entire expression is an integer, but it
+            #can be decomposed into fractions (i.e.
+            #numerator and denominator.
+            #The counters keep track of multiples of 2 in
+            #numerator and denominator to determine oddness.
+            count_denom = 0
+            count_num = 0
             for t in self.args:
-                if not t.is_integer:
-                    return None
-                elif t.is_even:
-                    r = False
-                elif t.is_integer:
-                    if r is False:
-                        pass
-                    elif acc != 1 and (acc + t).is_odd:
-                        r = False
-                    elif t.is_odd is None:
-                        r = None
-                acc = t
-            return r
+                if t == S.Half:
+                    count_denom += 1
+                elif isinstance(t,Symbol):
```

Comment:
I didn't know about '''self.as_coeff_mul()''', so I'll use this method as it would make it easier to evaluate.    I think the counters make sense as a way to keep track of the powers of 2 in the numerator, so I think I will keep that logic.  I'll try to find the function you were talking about regarding factoring the denominator.  Thanks again, you've been amazingly helpful!